core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -86,8 +86,14 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
-    if (!fileSystem.isDirectory(new Path(logBaseDir))) {
-      throw new IllegalArgumentException(s"Log directory $logBaseDir does not exist.")
+    try{
+        if(!fileSystem.getFileStatus(new Path(logBaseDir)).isDir()){
+          throw new IllegalArgumentException(s"Log directory $logBaseDir is not a directory.")
+        }
+        
+    }
+    catch{
+        case ioe: FileNotFoundException => fileSystem.mkdirs(new Path(logBaseDir))
```

Comment:
Let's at log a warning before creating this since as @vanzin points out, it may not have correct permissions for downstream consumers.